Core/Utilities/DEncode.py

```diff
@@ -16,236 +16,372 @@
 
 import types
 import datetime
+import os
+
+import inspect
+import traceback
+from pprint import pprint
+
+
+# Setting this environment variable to any value will enable the dump of the debugging
+# call stack
+DIRAC_DEBUG_DENCODE_CALLSTACK = bool(os.environ.get('DIRAC_DEBUG_DENCODE_CALLSTACK', False))
+
+# Depth of the stack to look for with inspect
+CONTEXT_DEPTH = 100
+
+
+def printDebugCallstack():
+  """ Prints information about the current stack as well as the caller parameters.
+      The purpose of this method is to track down all the places in DIRAC that might
+      not survive the change to JSON encoding.
+
+      :returns: None
+
+  """
+  def stripArgs(frame):
+    """ Keeps only the parameters and their values from a frame
+
+        :param frame: frame object
+
+        :returns: dict {param name: value}
+    """
+    # Get all the arguments of the call
+    allArgs = inspect.getargvalues(frame)
+    # Keep only the arguments that are parameters of the call, as well as their value
+    return dict([(argName, allArgs.locals[argName]) for argName in allArgs.args])
+
+  tb = traceback.format_stack()
+  frames = inspect.stack(context=CONTEXT_DEPTH)
+
+  # print the traceback that leads us here
+  # remove the last element which is the traceback module call
+  for line in tb[:-1]:
+    print line
+
+  # Now we try to navigate up to the caller of dEncode.
+  # For this, we find the frame in which we enter dEncode.
+  # We keep the parameters to display it.
+  # Then we navigate to the parent frame, and we display the file
+  # and line number where this call was done
+  try:
+    framesIter = iter(frames)
+    for frame in framesIter:
+      # First check that we are using either 'encode' or 'decode' function
+      if frame[3] in ('encode', 'decode'):
+        # Then check it is the good file
+        if frame[1].endswith('DIRAC/Core/Utilities/DEncode.py'):
+          # Keep the arguments of the DEncode call
+          dencArgs = stripArgs(frame[0])
+          # Take the calling frame
+          frame = next(framesIter)
+          print "Calling frame: %s" % (frame[1:3],)
+          print "With arguments ",
+          pprint(dencArgs)
+          break
+
+  except BaseException:
+    pass
+  print "=" * 100
+  print
+  print
+
 
 _dateTimeObject = datetime.datetime.utcnow()
-_dateTimeType = type( _dateTimeObject )
-_dateType = type( _dateTimeObject.date() )
-_timeType = type( _dateTimeObject.time() )
+_dateTimeType = type(_dateTimeObject)
+_dateType = type(_dateTimeObject.date())
+_timeType = type(_dateTimeObject.time())
 
 g_dEncodeFunctions = {}
 g_dDecodeFunctions = {}
 
-#Encoding and decoding ints
-def encodeInt( iValue, eList ):
-  eList.extend( ( "i", str( iValue ), "e" ) )
+# Encoding and decoding ints
```

Comment:
Maybe better to put this in the function docstring so autopep doesn't separate it from the function by two lines.

Reply:
Sure. I can do that for all of them